tests/importance_tests/test_init.py

```diff
@@ -103,6 +103,46 @@ def objective(trial: Trial) -> float:
         assert math.isclose(1.0, sum(i for i in param_importance.values()), abs_tol=1e-5)
 
 
+@parametrize_evaluator
+@parametrize_storage
+def test_get_param_importances_with_target(
+    storage_init_func: Callable[[], storages.BaseStorage],
+    evaluator_init_func: Callable[[], BaseImportanceEvaluator],
+) -> None:
+    def objective(trial: Trial) -> float:
+        x1 = trial.suggest_uniform("x1", 0.1, 3)
+        x2 = trial.suggest_loguniform("x2", 0.1, 3)
+        x3 = trial.suggest_discrete_uniform("x3", 0, 3, 1)
+        if trial.number % 2 == 0:
+            x4 = trial.suggest_uniform("x4", 0.1, 3)
+
+        value = x1 ** 4 + x2 + x3
+        if trial.number % 2 == 0:
+            value += x4
+        return value
+
+    study = create_study(storage_init_func())
+    study.optimize(objective, n_trials=100)
+
+    param_importance = get_param_importances(
+        study,
+        evaluator=evaluator_init_func(),
+        target=lambda t: t.params["x1"] + t.params["x2"],
+    )
+
+    assert isinstance(param_importance, OrderedDict)
+    assert len(param_importance) == 3
+    assert all(param_name in param_importance for param_name in ["x1", "x2", "x3"])
+    prev_importance = float("inf")
+    for param_name, importance in param_importance.items():
+        assert isinstance(param_name, str)
+        assert isinstance(importance, float)
+        assert importance <= prev_importance
+        prev_importance = importance
+    assert param_importance["x3"] == min(param_importance.values())
```

Comment:
This is indeed expected but I'm not sure it's suitable as a unit test without fixing the seed. It's also more likely to fail if we reduce the number of trials as suggested by @HideakiImamura. Since the current interface of the evaluator makes it a bit difficult to do this, I think we can omit this line from this general HPI test.

Reply:
Yes, this assertion actually fails when we reduce `n_trials`, so I chose large `n_trials`. I think it is not suitable for a unit test. I agree to omit it.